### PR TITLE
Fixed: #590 add emojipedia play

### LINKED ISSUE #590: [Add a Play]: Emojipedia Play
Thank You! 

## What's the Play about?
It's a fun play where user can search for any emojis and can copy it.

## What ReactJS concept will be used to create this play?
I'll be using Reactjs hooks, API, load more feature etc

## Are you willing to take it up for implementation?
Yes, I would love to work on this

### PR DESCRIPTION
**Pull request by makdoom**
# Description

Emojipedia is a fun project play where users can search for any emojis and can copy or use it anywhere.

In this project I've used [Open Emoji API](https://emoji-api.com/). So the environment variable REACT_APP_EMOJI_API_KEY is required.

Fixes #590

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tested locally.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/plays/emojipedia/Emojipedia.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    if ("clipboard" in navigator) {
+      await navigator.clipboard.writeText(emojiCharacter);
+      toast.success(`Emoji ${emojiCharacter} copied to clipboard.`);
+    }
+  };
+
+  // Search filter
+  useEffect(() => {
+    const delayFn = setTimeout(() => {
+      if (query !== "") {
```
Comment (harshsinghatz) [inline src/plays/emojipedia/Emojipedia.js:34]: Here you can directly write:
`if (!query)`

**Comment by atapas** [comment]
@Sachin-chaurasiya please review..

@harshsinghatz feel free to review more.

File: src/plays/emojipedia/Emojipedia.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        setEmojisList(
+          emojisList?.filter((emoji) =>
+            emoji?.unicodeName?.toLowerCase().includes(query?.toLowerCase())
+          )
+        );
```
Comment (Sachin-chaurasiya) [inline src/plays/emojipedia/Emojipedia.js:39]:
```suggestion
        setEmojisList(previousList => previousList.filter((emoji) =>
            emoji?.unicodeName?.toLowerCase().includes(query?.toLowerCase())
          ));

```

File: src/plays/emojipedia/Emojipedia.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    return () => clearTimeout(delayFn);
+  }, [query, data, emojisList]);
+
+  useEffect(() => {
+    if (error) return toast.error(error);
+
+    // Set Emoji list
+    if (data?.length) setEmojisList(data);
+
```
Comment (Sachin-chaurasiya) [inline src/plays/emojipedia/Emojipedia.js:53]:
```suggestion
    if (error) {
     toast.error(error);
    }

    // Set Emoji list
    if (data?.length) {
    setEmojisList(data);
    }


```

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]

File: src/plays/emojipedia/Emojipedia.js (hunk 1 above)
Comment (makdoom) [reply src/plays/emojipedia/Emojipedia.js:34]: done 👍

File: src/plays/emojipedia/Emojipedia.js (hunk 2 above)
Comment (makdoom) [reply src/plays/emojipedia/Emojipedia.js:39]: Done

File: src/plays/emojipedia/Emojipedia.js (hunk 3 above)
Comment (makdoom) [reply src/plays/emojipedia/Emojipedia.js:53]: done

**Review by Sachin-chaurasiya — APPROVED** [review]

**Comment by makdoom** [comment]
hey @atapas here we need to add API_KEY from [Open Emoji API](https://emoji-api.com/) by the name REACT_APP_EMOJI_API_KEY in vercel environment

**Comment by makdoom** [comment]
hey @atapas I've made all the necessary changes, any update on this PR ?

**Comment by atapas** [comment]
> hey @atapas here we need to add API_KEY from [Open Emoji API](https://emoji-api.com/) by the name REACT_APP_EMOJI_API_KEY in vercel environment

Added

**Review by atapas — APPROVED** [review]
@makdoom Accepted

**Pull request merged**